test/mjsunit/wasm/asm-wasm.js

```diff
@@ -267,4 +267,70 @@ function TestContinueInNamedWhile() {
   return {caller: caller};
 }
 
-assertEquals(20, WASM.asmCompileRun(TestContinueInNamedWhile.toString()));
\ No newline at end of file
+assertEquals(20, WASM.asmCompileRun(TestContinueInNamedWhile.toString()));
+
+function TestNot() {
+  "use asm";
+
+  function caller() {
+    var a = !(2 > 3);
+    return a | 0;
+  }
+
+  return {caller:caller};
+}
+
+assertEquals(1, WASM.asmCompileRun(TestNot.toString()));
+
+function TestNotEquals() {
+  "use asm";
+
+  function caller() {
+    var a = 3;
+    if (a != 2) {
+      return 21;
+    }
+    return 0;
+  }
+
+  return {caller:caller};
+}
+
+assertEquals(21, WASM.asmCompileRun(TestNotEquals.toString()));
+
+function TestUnsignedComparison() {
+  "use asm";
+
+  function caller() {
+    var a = 0xffffffff;
+    if ((a>>>0) > (0>>>0)) {
+      return 22;
+    }
+    return 0;
+  }
+
+  return {caller:caller};
+}
+
```

Comment:
I'll check into this soon.
Can you comment these out with a TODO(bradnelson): enable when typer is updated?